service/controller/v14/resource/endpoint/current.go

```diff
@@ -20,33 +17,20 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 
 	var serviceName string
 	{
-		r.logger.LogCtx(ctx, "level", "debug", "message", "looking for annotations on pod")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding annotations")
 
 		_, serviceName, err = getAnnotations(*pod, IPAnnotation, ServiceAnnotation)
 		if IsMissingAnnotationError(err) {
-			r.logger.LogCtx(ctx, "level", "debug", "message", "annotation is missing on pod")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "did not find annotations")
 			resourcecanceledcontext.SetCanceled(ctx)
 			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
 
 			return nil, nil
 		} else if err != nil {
 			return nil, microerror.Mask(err)
 		}
-	}
 
-	if key.IsPodDeleted(pod) {
-		isDrained, err := key.IsPodDraind(pod)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
-		if !isDrained {
-			r.logger.LogCtx(ctx, "level", "debug", "message", "cannot finish deletion of pod due to undrained status")
-			resourcecanceledcontext.SetCanceled(ctx)
-			finalizerskeptcontext.SetKept(ctx)
-			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-
-			return nil, nil
-		}
```

Comment:
Removing this fixes https://github.com/giantswarm/giantswarm/issues/3782. I cannot recall why we added it, but it prevents terminating pods from getting their broken IPs being removed from endpoints. This works now. The API accessibility issues we perceived as described in the postmortems are all gone now. I am very pleased with this.

Reply:
I don't follow there. This is something based on annotations I don't really remember. It's also used in endpoint resource.

Reply:
This here is the endpoint resource.

Reply:
LoL :D 
OK, so these functions should be also removed from key package.

Reply:
Good catch. Removed `IsPodDeleted` from key package. `IsPodDraind` is still used in the `pod` resource which we use for draining.